common.blocks/i-bem-dom/__event-manager/_type/i-bem-dom__event-manager_type_bem.js

```diff
@@ -0,0 +1,231 @@
+                    e,
+                    params.selector,
+                    data,
+                    fnStorage[fnId] = this._fnWrapper(fn, fnId));
+            }
+
+            return this;
+        },
+
+        /**
+         * Removes event handler or handlers
+         * @param {String} [e] Event type
+         * @param {Function} [fn] Handler
+         * @returns {EventManager} this
+         */
+        un : function(e, fn) {
+            if(typeof e === 'object') {
+                objects.each(e, function(fn, e) {
+                    this.un(e, fn);
+                }, this);
+            } else {
+                var argsLen = arguments.length;
+                if(!argsLen) {
+                    objects.each(this._storage, this._unbindByEvent, this);
+                } else {
+                    var params = this._params;
+
+                    e = buildEventName(e, params.entityCls.getEntityName());
+
+                    if(argsLen === 1) {
+                        this._unbindByEvent(this._storage[e], e);
+                    } else {
+                        var wrappedFn;
+                        if(params.entityCls) {
+                            var fnId = identify(fn),
+                                fnStorage = this._storage[e];
+                            wrappedFn = fnStorage && fnStorage[fnId];
+                            wrappedFn && (fnStorage[fnId] = null);
+                        }
+                        params.ctx.off(e, params.selector, wrappedFn || fn);
+                    }
+                }
+            }
+
+            return this;
+        },
+
+        _unbindByEvent : function(fnStorage, e) {
+            var params = this._params;
+            fnStorage && objects.each(fnStorage, function(fn) {
+                params.ctx.off(e, params.selector, fn);
+            });
+            this._storage[e] = null;
+        }
+    });
+
+provide({
+    getInstanceEventManager : function(instance, ctx, getEntityCls) {
+        var instanceId = identify(instance),
+            instanceStorage = eventStorage[instanceId];
+
+        if(!instanceStorage) {
+            instanceStorage = eventStorage[instanceId] = {};
+            instance.on({ modName : 'js', modVal : '' }, function() {
+                delete eventStorage[instanceId];
+            });
+        }
+
+        var params = buildBemEventManagerParams(
+                ctx,
+                instance.domElem,
+                '',
+                instance.__self,
+                getEntityCls);
+
+        return instanceStorage[params.key] ||
+            (instanceStorage[params.key] = new EventManager(
+                params,
+                function(fn, fnId) {
+                    return function(e, data, flags, originalEvent) {
+                        if(!flags[fnId]) {
+                            originalEvent.data = e.data;
+                            originalEvent.bemTarget = $(e.currentTarget).bem(params.entityCls);
```

Comment:
Can we avoid jquery usage in event managers? ;-(

Can we use `i-bem-dom` itself for typecasting instead of 3rd-party (even if global and popular) libraries?